### PR TITLE
Connect to mongo db 14

### PR DESCRIPTION
**Pull request by MohamadShaheen**

### CODE REVIEW

File: DAL/basic_db_functions.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import random
+from pymongo import MongoClient
+from db.connection_credentials import mongodb_url
+
+
+class BasicDbFunctions:
```
Comment (yishain11) [inline DAL/basic_db_functions.py:6]: is there a special reason to use a class here?

File: routes/basic_db_functions_route.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from fastapi import APIRouter
+from DAL.basic_db_functions import BasicDbFunctions
+
+router = APIRouter()
+
+
+@router.get('/')
```
Comment (yishain11) [inline routes/basic_db_functions_route.py:7]: if the point of this endpoint is to store some data in mong, your endpoint name and function name need to be better.

File: routes/basic_db_functions_route.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from fastapi import APIRouter
+from DAL.basic_db_functions import BasicDbFunctions
+
+router = APIRouter()
+
```
Comment (yishain11) [inline routes/basic_db_functions_route.py:5]: remove 1 line space

File: routes/basic_db_functions_route.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+@router.get('/')
+async def test():
+    basic_db_functions = BasicDbFunctions()
+    stored_number = basic_db_functions.test()
+
```
Comment (yishain11) [inline routes/basic_db_functions_route.py:11]: remove line space

File: server.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from fastapi import FastAPI
+from routes import basic_db_functions_route
+
+app = FastAPI()
+
```
Comment (yishain11) [inline server.py:5]: remove exssesive line spaces!

**Review by yishain11 — CHANGES_REQUESTED** [review]
what is inside the DAL folder? what name is it? will you remember in 2 weeks what DAL stands for? change it

also i the function name "test" is too generic and not descriptive. what are you testing?
in one month, if i ask you what does the function "test" do, would you be able to answer?

**Comment by MohamadShaheen** [comment]
> what is inside the DAL folder? what name is it? will you remember in 2 weeks what DAL stands for? change it
> 
> also i the function name "test" is too generic and not descriptive. what are you testing? in one month, if i ask you what does the function "test" do, would you be able to answer?

DAL stands for Data Access Layer and yes I would remember it after 1 year from now, I don’t want to write such long directory name.

Plus, the test command will be removed later anyway, it was made for the purpose of this issue and will be removed anyway later so there won’t be any harm from having it now.

File: DAL/basic_db_functions.py (hunk 1 above)
Comment (MohamadShaheen) [reply DAL/basic_db_functions.py:6]: I could refrain from using a class here, it’s just a convention I follow, connection with databases is done by classes. That’s the convention I follow and followed for my previous project.

File: routes/basic_db_functions_route.py (hunk 2 above)
Comment (MohamadShaheen) [reply routes/basic_db_functions_route.py:7]: As said before, this endpoint will be removed later anyway, so there is no harm from accepting it.

File: routes/basic_db_functions_route.py (hunk 3 above)
Comment (MohamadShaheen) [reply routes/basic_db_functions_route.py:5]: I use PyCharm format option for formatting files and it doesn’t give me any warning when loading those files.

File: server.py (hunk 5 above)
Comment (MohamadShaheen) [reply server.py:5]: I do it intentionally for having a better look for the code, it’s more readable as from my point of view when I leave line spaces. I deal with code as chunks, when I finish a specific chunk I intentionally leave a line space to announce a start of a new chunk.

Comment (MohamadShaheen) [reply server.py:5]: That’s also what I see all youtubers and coders do, and also any code that ChatGPT provides, which means it’s a practice among coders.

File: tests/test_basic_db_functions.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from fastapi.testclient import TestClient
+from server import app
+import re
+
+client = TestClient(app)
+
+def test_test_store_in_mongodb():
```
Comment (yishain11) [inline tests/test_basic_db_functions.py:7]: this is not the right way to test for this.
if you want to test insert, you need to do 2 things:
1. load the data and see if the new value is inside
2. after the test is complete - delete the new data, to leave no side effects

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Pull request merged**